dns/zone.py

```diff
@@ -500,16 +501,27 @@ def to_file(self, f, sorted=True, relativize=True, nl=None):
 
         str_type = string_types
 
-        if nl is None:
-            opts = 'wb'
-        else:
-            opts = 'wb'
-
         if isinstance(f, str_type):
-            f = open(f, opts)
+            f = open(f, 'wb')
             want_close = True
         else:
             want_close = False
+
+        # must be in this way, f.encoding may contain None, or even attribute
+        # may not be there
+        file_enc = getattr(f, 'encoding', None)
+        if file_enc is None:
+            file_enc = 'utf-8'
+
+        if nl is None:
+            nl_b = os.linesep.encode(file_enc)  # binary mode, '\n' is not enough
+            nl = u'\n'
```

Comment:
IIRC in text mode '\n' should be transformed to platform EOL.

Please read https://docs.python.org/2/library/os.html#os.linesep (there is note to not use linesep in textual mode)